src/main/java/uk/gov/hmcts/ccd/domain/service/FindUserProfileOperation.java

```diff
@@ -18,9 +28,35 @@ public FindUserProfileOperation(UserProfileRepository userProfileRepository) {
     }
 
     public UserProfile execute(String userProfileId, String actionedBy) {
-        final UserProfile userProfile = userProfileRepository.findById(userProfileId, actionedBy);
+        String decodedUid = UriUtils.decode(userProfileId, "UTF-8");
+        final UserProfile userProfile = userProfileRepository.findById(decodedUid, actionedBy);
         return Optional.ofNullable(userProfile)
             .orElseThrow(() -> new NotFoundException("Cannot find user profile"));
     }
+
+    public UserProfileCollection execute(List<String> userProfileIds, String actionedBy) {
+        validateRequest(userProfileIds);
+        final List<UserProfile> userProfiles = userProfileIds.stream()
+            .map(userProfileId -> userProfileRepository.findById(userProfileId, actionedBy))
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+        return new UserProfileCollection(userProfiles);
+    }
+
+    private void validateRequest(List<String> userProfileIds) {
+        if (userProfileIds == null || userProfileIds.isEmpty()
+            || userProfileIds.stream().anyMatch(emailId -> !isValidEmailAddress(emailId))) {
+            throw new BadRequestException("Email Id(s) not valid");
+        }
+    }
+
+    private boolean isValidEmailAddress(final String emailId) {
+        try {
+            new InternetAddress(emailId).validate();
```

Comment:
we need to be very careful here - making sure the standard asserted here accepts all the email addresses we have on prod. Otherwise, we may have a lighter check on email format.

Reply:
Good point. We didn't have any email validation in the previous API version. I just introduced a new application param to enable the email validation. By default, the email validation will be disabled.